cli/execute.py

```diff
@@ -59,96 +39,57 @@ def ansible_playbook(playbook, args, inventory="local_hosts"):
-
-    runner_cb = callbacks.PlaybookRunnerCallbacks(
-        stats,
-        verbose=ansible.utils.VERBOSITY
-    )
-
-    module_path = CONF.get('defaults', 'modules')
-
-    pb = ansible.playbook.PlayBook(
-        # From ansible-playbook:
-        playbook=path_to_playbook,
-        inventory=ansible.inventory.Inventory(inventory),
-        extra_vars=args['settings'],
-        callbacks=playbook_cb,
-        runner_callbacks=runner_cb,
-        stats=stats,
-        module_path=module_path
     )
-
-    failed_hosts = []
-    unreachable_hosts = []
-
-    if args['verbose']:
-        ansible_cmd = ["ansible-playbook"]
-        if module_path:
-            ansible_cmd.append("-M " + module_path)
-        ansible_cmd.append("-" + "v" * args['verbose'])
-        ansible_cmd.append("-i " + inventory)
-        extra_vars = args['output'] or "<path to settings file>"
-        ansible_cmd.append("--extra-vars @" + extra_vars)
-        ansible_cmd.append(path_to_playbook)
-        print "ANSIBLE COMMAND: " + " ".join(ansible_cmd)
-
-    pb.run()
-
-    hosts = sorted(pb.stats.processed.keys())
-    callbacks.display(callbacks.banner("PLAY RECAP"))
-    playbook_cb.on_stats(pb.stats)
-
-    for host in hosts:
-        status = pb.stats.summarize(host)
-        if status['failures'] > 0:
-            failed_hosts.append(host)
-        if status['unreachable'] > 0:
-            unreachable_hosts.append(host)
-
-    retries = failed_hosts + unreachable_hosts
-
-    if len(retries) > 0:
-        filename = pb.generate_retry_inventory(retries)
-        if filename:
-            callbacks.display(
-                "           to retry, use: --limit @%s\n" % filename)
-
-    for host in hosts:
-        status = pb.stats.summarize(host)
-
-        callbacks.display("%s : %s %s %s %s" % (
-            hostcolor(host, status),
-            colorize('ok', status['ok'], 'green'),
-            colorize('changed', status['changed'], 'yellow'),
-            colorize('unreachable', status['unreachable'], 'red'),
-            colorize('failed', status['failures'], 'red')), screen_only=True)
-
-        callbacks.display("%s : %s %s %s %s" % (
-            hostcolor(host, status, False),
-            colorize('ok', status['ok'], None),
-            colorize('changed', status['changed'], None),
-            colorize('unreachable', status['unreachable'], None),
-            colorize('failed', status['failures'], None)), log_only=True)
-
-    print ""
-    if len(failed_hosts) > 0:
-        raise exceptions.IRPlaybookFailedException(
-            playbook, "Failed hosts: %s" % failed_hosts)
-    if len(unreachable_hosts) > 0:
-        raise exceptions.IRPlaybookFailedException(
-            playbook, "Unreachable hosts: %s" % unreachable_hosts)
+    options = namedtuple('Options', hacked_options.keys())(**hacked_options)
+
+    passwords = dict(vault_pass='secret')
+    variable_manager = VariableManager()
+    variable_manager.extra_vars = args["settings"]
```

Comment:
Is the 'settings' key is standard to ansible 2.0. I think we need to document the structure of the args parameter.